### PR TITLE
Migrate Windows CI to GitHub Actions.

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

It is quite difficult to maintain current Windows CI, because it uses pre-compiled dependencies stored on wolt.com server and there is low chance to find someone who is able to update it.

Needed for https://github.com/Project-OSRM/osrm-backend/pull/6300

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: scripts/ci/windows-build.bat
Comment (SiarheiFedartsou) [inline scripts/ci/windows-build.bat:1]: It is modified old `appveyor-build.bat`

File: CMakeLists.txt
Comment (SiarheiFedartsou) [inline CMakeLists.txt:487]: Tbh I spend several days trying to understand why `osrm-extract.exe` silently finishes with `-1073741515` exit code(Google says that it is something about DLL loading), then I found that it happens on the first access of any TBB symbol. I tried to statically compile it and it worked. I believe it is possible to make it working with dynamic linkage, but I have no Windows machine and had to debug all these things directly on CI, so debugging is really limited for me.

File: CMakeLists.txt
Comment (SiarheiFedartsou) [inline CMakeLists.txt:448]: https://stackoverflow.com/questions/70898030/boost-link-error-using-conan-find-package

**Comment by SiarheiFedartsou** [comment]
@mjjbell please take a look when you have a chance

**Comment by SiarheiFedartsou** [comment]
Also probably someone with corresponding access rights should disable integration with AppVeyor:
Screenshot 2022-08-16 at 23 45 25

File: CMakeLists.txt
Comment (mjjbell) [reply CMakeLists.txt:487]: :+1: 
Can you add a comment to that effect in the file?

File: CMakeLists.txt
Comment (mjjbell) [reply CMakeLists.txt:448]: Let's add this to the file too, for future reference.

**Comment by mjjbell** [comment]
I don't have access to the status checks. However I have access to the appveyor project, so I've removed the integration from that side.

File: CMakeLists.txt
Comment (SiarheiFedartsou) [reply CMakeLists.txt:491]: Added.

**Comment by mjjbell** [comment]
> I don't have access to the status checks. However I have access to the appveyor project, so I've removed the integration from that side.

I think we still need someone to disable it in the repo. @danpat @TheMarex would you be able to disable the Appveyor check in the repo settings?

File: .github/workflows/osrm-backend.yml
Comment (mjjbell) [inline .github/workflows/osrm-backend.yml:23]: I forgot about this with the Docker job.

[ci-complete](https://github.com/Project-OSRM/osrm-backend/pull/6312/files#diff-4f680fa9a2e9b55e339a5fc727a44e968318e00f54beb55da0e5c5b24bbd2f9dR642) is the meta-job that PRs will block merging on. We should add `windows` and `docker` to the list of `needs`, in the unlikely case that all the Linux/Mac jobs succeed before those two complete/fail.

Comment (SiarheiFedartsou) [reply .github/workflows/osrm-backend.yml:23]: Okay, will do. Btw while we are waiting for someone to disable Appveyor check, may be just return that job and make it no-op(i.e. always green?). It would unblock this PR from being merged and would allow to work on other PRs which depend on it.

Comment (mjjbell) [reply .github/workflows/osrm-backend.yml:23]: Good idea :+1:

**Review by mjjbell — APPROVED** [review]
LGTM :rocket: 

One step closer to being able to publish Windows releases too.

**Comment by DennisOSRM** [comment]
Removed the appveyor web hook

**Pull request merged**
